root@amsix:/home/dolf/Projects/ZF-Standard-Trunk/library/Zend# svn diff Acl.php          
Index: Acl.php                                                                           
===================================================================                      
--- Acl.php     (revision 16392)                                                         
+++ Acl.php     (working copy)                                                           
@@ -1023,12 +1023,23 @@
             }
             return $visitor['allRoles'];
         }
+
+
         $roleId = $role->getRoleId();
         if (!isset($visitor['byRoleId'][$roleId])) {
-            if (!$create) {
-                return $nullRef;
+               if (!$create) {
+                       if(isset($this->_rules['allResources']['byRoleId'][$roleId])) {
+                               var_dump($visitor);
+                               $out = $visitor+$this->_rules['allResources']['byRoleId'][$roleId];
+                               $outRef = &$out;
+                               return $outRef;
+                               return $nullRef;
+                       } else {
+                    return $nullRef;
+                       }
+            } else {
+                $visitor['byRoleId'][$roleId]['byPrivilegeId'] = array();
             }
-            $visitor['byRoleId'][$roleId]['byPrivilegeId'] = array();
         }
         return $visitor['byRoleId'][$roleId];
     }
